predicators/envs/base_env.py

```diff
@@ -176,6 +181,19 @@ def get_test_tasks(self) -> List[EnvironmentTask]:
                 self._test_tasks = self._generate_test_tasks()
         return self._test_tasks
 
+    @property
+    def _current_state(self) -> State:
+        """Default for environments where states are observations."""
+        assert isinstance(self._current_observation, State)
+        return self._current_observation
+
+    def goal_reached(self) -> bool:
+        """Default implementation assumes environment tasks are tasks.
+
+        Subclasses may override.
+        """
+        return self._current_task.task.goal_holds(self._current_state)
```

Comment:
Following my comment on the `EnvironmentTask` PR, I still feel like CogMan should at least produce the state, if not the actual goal as well. Maybe we can move the dummy methods in `EnvironmentTask` into CogMan?